interactions/client.py

```diff
@@ -132,14 +132,34 @@ async def __compare_sync(self, data: dict, pool: List[dict]) -> bool:
         :return: Whether the command has changed or not.
         :rtype: bool
         """
-        attrs: List[str] = ["type", "name", "description", "options", "guild_id"]
+        attrs: List[str] = [
```

Comment:
So, I was thinking on this months back when I was refactoring the whole sync system. How would you feel if we did this instead?

```py
attrs: List[str] = [name for name in ApplicationCommand.__dict__.keys() if not name.startswith("_")]
```

This would simply generate a whole list of attributes to check for if they're not dundered. This will obviously ignore `_json`, and (if potentially added in the future) `_client`. Additionally, since our actual comparison condition checks for if the attribute name exists in the pool of data we're looking for, we don't have to worry about whether we're excessively checking for certain attributes or not. It also avoids us having to constantly update the list of attributes every time Discord releases new features to the application command object, as we only have to update the type annotation in the class itself. (Pretty much avoids WET code)

Reply:
Yeah, I'll do that, but is there a benefit of using `__dict__` instead of just `__slots__`?

Reply:
I'm not sure if `__slots__` can actually be read from in terms of reading data. If they can, then no benefit besides preference of how you're going about it.

Reply:
Done